packages/ui5-application-writer/templates/optional/codeAssist/package.json

```diff
@@ -2,6 +2,6 @@
     "devDependencies": {
         "eslint": "7.32.0",
         "@sap/eslint-plugin-ui5-jsdocs": "2.0.5",
-        "@sapui5/ts-types": "1.92.2"
+        "@sapui5/ts-types-esm": "latest"
```

Comment:
I read https://github.com/SAP/ui5-typescript#how-to-obtain-the-ui5-typescript-signatures and understood the `-esm` is to be used and the other discouraged. 
@petermuessig could you please clarify?

Reply:
This option is for code assist with JS project rather than TS. But not sure what is the recommended in this case.

Reply:
When using TypeScript the `@sapui5/ts-types-esm` must be used! When using classic JavaScript the `@sapui5/ts-types` can be used. This is optional also because the JavaScript types aren't as good as the esm types we provide for TypeScript development.

Reply:
https://github.com/SAP/open-ux-tools/pull/667/commits/2d410317d29dc54a154ab7a6bf273fed0565c677

Reply:
Thanks @tobiasqueck @petermuessig

Reply:
Roughly as Peter wrote:
a [section in the main TS documentation](https://sap.github.io/ui5-typescript/#how-to-profit-from-typescript-even-when-doing-plain-javascript-development) links to a tutorial about using the type definitions for pure JS development. In particular [this section](https://github.com/SAP-samples/ui5-cap-event-app/blob/js-with-typescript-support/README.md#add-needed-dependencies) discusses the advantages and disadvantages of using the “esm” types. In short: *you can use either; using the esm types in JS looks a bit uglier, but the quality focus is on them*:

> NOTE: we are working with the new "-esm" flavor of the type definitions. This has advantages but also disadvantages when developing in JavaScript, so you might also choose the @sapui5/ts-types package instead:
> •	Advantage: the ts-types-esm packages are the best-quality type definition files, the ones the UI5 team is focusing on. While the ts-types packages are also automatically updated with every new UI5 release, there might be some TypeScript glitches inside which are not fixed with priority.
> •	Disadvantage: The "-esm" definitions define only ES6 modules, not the legacy global names of UI5 classes (like "sap.m.Button" in the global namespace). This makes referencing types in the JSDoc comments a bit ugly. Instead of @type {sap.m.Button} one has to write @type {import('sap/m/Button').default}